nomad/structs/structs.go

```diff
@@ -11428,6 +11443,59 @@ func (a *Allocation) NeedsToReconnect() bool {
 	return disconnected
 }
 
+// GetLeaderTask will return the leader task in the allocation
+// if there is one, otherwise it will return an empty task.
+func (a *Allocation) GetLeaderTask() Task {
+	tg := a.Job.LookupTaskGroup(a.TaskGroup)
+
+	task := Task{}
+
+	if tg == nil {
+		return task
+	}
+
+	switch len(tg.Tasks) {
+	case 0:
+		return task
+
+	case 1:
+		task = *tg.Tasks[0]
+
+	default:
+		for _, t := range tg.Tasks {
+			if t.Leader {
+				task = *t
+				break
+			}
+		}
+	}
+
+	return task
+}
+
+// LatestStartOfTask returns the time of the last start event for the given task
+// using the allocations TaskStates. If the task has not started, the zero time
+// will be returned.
+func (a *Allocation) LatestStartOfTask(taskName string) time.Time {
```

Comment:
```suggestion
// LastTaskStart returns the time of the last start event for the given task
// using the allocations TaskStates. If the task has not started, the zero time
// will be returned.
func (a *Allocation) LastTaskStart(taskName string) time.Time {
```

Another nit-pick 😅 

This name matches the name of `LastUnknown`, which returns a similar result. It could also be useful to move the method so it's closer to that one.
https://github.com/hashicorp/nomad/blob/3193ac204f6564711004e00948e43146ce1399c4/nomad/structs/structs.go#L11394-L11409